std-bits/base/src/main/java/org/enso/base/enso_cloud/audit/AuditLogApiAccess.java

```diff
@@ -0,0 +1,220 @@
+package org.enso.base.enso_cloud.audit;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+import org.enso.base.enso_cloud.AuthenticationProvider;
+import org.enso.base.enso_cloud.CloudAPI;
+
+/**
+ * Gives access to the low-level log event API in the Cloud and manages asynchronously submitting
+ * the logs.
+ */
+class AuditLogApiAccess {
+  private static final Logger logger = Logger.getLogger(AuditLogApiAccess.class.getName());
+
+  /**
+   * We still want to limit the batch size to some reasonable number - sending too many logs in one
+   * request could also be problematic.
+   */
+  private static final int MAX_BATCH_SIZE = 100;
+
+  private static final int MAX_RETRIES = 5;
+
+  public static AuditLogApiAccess INSTANCE = new AuditLogApiAccess();
+
+  private HttpClient httpClient;
+  private final LogJobsQueue logQueue = new LogJobsQueue();
+  private volatile RequestConfig requestConfig = null;
+  private final ThreadPoolExecutor backgroundThreadService;
+
+  private AuditLogApiAccess() {
+    // We set-up a thread 'pool' that will contain at most one thread.
+    // If the thread is idle for 60 seconds, it will be shut down.
+    backgroundThreadService =
+        new ThreadPoolExecutor(0, 1, 60L, TimeUnit.SECONDS, new LinkedBlockingQueue<>());
+  }
+
+  public Future<Void> logWithConfirmation(LogMessage message) {
+    CompletableFuture<Void> completionNotification = new CompletableFuture<>();
+    enqueueJob(new LogJob(message, completionNotification));
+    return completionNotification;
+  }
+
+  public void logWithoutConfirmation(LogMessage message) {
+    enqueueJob(new LogJob(message, null));
+  }
+
+  private void enqueueJob(LogJob job) {
+    ensureConfigSaved();
+    int queuedJobs = logQueue.enqueue(job);
+    if (queuedJobs == 1 && backgroundThreadService.getQueue().isEmpty()) {
```

Comment:
I wasn't sure about this `queuedJobs == 1` check, but when I realized `logQueue.enqueue` is synchronized method, I believe it is going to work properly.